packages/runtime/container-runtime/src/deltaScheduler.ts

```diff
@@ -25,81 +29,111 @@ import {
+            if (this.schedulingLog === undefined) {
                 if (this.schedulingCount % 2000 === 0) {
+                    // Every 2000th time we are scheduling the inbound queue, we log telemetry for the
+                    // number of ops processed, the time and number of turns it took to process the ops.
                     this.schedulingLog = {
-                        numberOfOps: this.deltaManager.inbound.length,
+                        opsRemainingToProcess: this.deltaManager.inbound.length,
                         numberOfTurns: 1,
                         totalProcessingTime: 0,
                     };
                 }
             }
 
-            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
-            const elapsedTime = performance.now() - this.processingStartTime!;
-            if (elapsedTime > this.totalProcessingTime) {
-                // We have processed ops for more than the total processing time. So, pause the
-                // queue, yield the thread and schedule a resume.
-
-                // eslint-disable-next-line @typescript-eslint/no-floating-promises
-                this.deltaManager.inbound.pause();
-                setTimeout(() => {
-                    this.deltaManager.inbound.resume();
-                });
-
-                this.processingStartTime = undefined;
-                // Increase the total processing time. Keep doing this after each turn until all the ops have
-                // been processed. This way we keep the responsiveness at the beginning while also making sure
-                // that all the ops process fairly quickly.
-                this.totalProcessingTime += this.processingTimeIncrement;
-
-                // If we are logging the telemetry this time, update the telemetry log object.
-                if (this.schedulingLog) {
-                    this.schedulingLog.numberOfTurns++;
-                    this.schedulingLog.totalProcessingTime += elapsedTime;
+            if (this.shouldRunScheduler()) {
+            {
+                const currentTime = performance.now();
+                // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+                const elapsedTime = currentTime - this.processingStartTime!;
+                if (elapsedTime > this.totalProcessingTime) {
+                    // We have processed ops for more than the total processing time. So, pause the
+                    // queue, yield the thread and schedule a resume.
+
+                    // eslint-disable-next-line @typescript-eslint/no-floating-promises
+                    this.deltaManager.inbound.pause();
+
+                    setTimeout(() => {
+                        if (this.schedulingLog) {
+                            this.logger.sendTelemetryEvent({
+                                eventName: "DeltaManagerPaused",
+                                duration: TelemetryLogger.formatTick(elapsedTime),
```

Comment:
The names for various times logged here are not clear.
- Do we need duration aka elaspedTime? That is the time spent in processing the last batch not the total time spent in processing a set of ops until we ran out of time.
- I think logging this.schedulingLog.totalProcessingTime should be enough. Although, we should update it to add elaspedTime before we log this message.
- waitingToResume -> timeSpentWhenPaused or timeToResume? Basically, indicate that this is a time.
Also, we should add `ms` suffix to all time related properties. That will make it easier to understand when looking at telemetry.